src/universal_transfer_operator/data_providers/database/google/bigquery.py

```diff
@@ -155,6 +238,208 @@ def truncate_table(self, table):
         """Truncate table"""
         self.run_sql(f"TRUNCATE {self.get_table_qualified_name(table)}")
 
+    def is_native_autodetect_schema_available(  # skipcq: PYL-R0201
+        self, file: File  # skipcq: PYL-W0613
+    ) -> bool:
+        """
+        Check if native auto-detection of schema is available.
+
+        :param file: File used to check the file type of to decide whether there is a native auto-detection available.
+        """
+        supported_config = self.NATIVE_AUTODETECT_SCHEMA_CONFIG.get(file.location.location_type)
+        if supported_config and file.type.name in supported_config["filetype"]:  # type: ignore
+            return True
+        return False
+
+    def create_table_using_native_schema_autodetection(  # skipcq: PYL-R0201
+        self,
+        table: Table,
+        file: File,
+    ) -> None:
+        """
+        Create a SQL table, automatically inferring the schema using the given file via native database support.
+        :param table: The table to be created.
+        :param file: File used to infer the new table columns.
+        """
+        supported_config = self.NATIVE_AUTODETECT_SCHEMA_CONFIG.get(file.location.location_type)
+        return supported_config["method"](table=table, file=file)  # type: ignore
+
+    # Require skipcq because method overriding we need param target_table
+    def is_native_path_available(  # skipcq: PYL-R0201
+        self,
+        source_dataset: File | Table,  # skipcq PYL-W0613
+    ) -> bool:
+        """
+        Check if there is an optimised path for source to destination.
+        :param source_file: File from which we need to transfer data
+        :param target_table: Table that needs to be populated with file data
+        """
+        if isinstance(source_dataset, Table):
+            return False
+        file_type = NATIVE_PATHS_SUPPORTED_FILE_TYPES.get(source_dataset.type.name)
+        location_type = self.NATIVE_PATHS.get(source_dataset.location.location_type)
+        return bool(location_type and file_type)
+
+    def load_file_to_table_natively(
+        self,
+        source_file: File,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        **kwargs,
+    ):
+        """
+        Checks if optimised path for transfer between File location to database exists
```

Comment:
Based on the docstring, I see we're checking location type, but do we not need to also check the file type to identify if an optimised path exists or not?

Reply:
Yes, we already do that. We have mapping of both filetype and location type as per the following:

https://github.com/astronomer/apache-airflow-provider-transfers/blob/502a76a396e543847c2327e5e51861abc4854f12/src/universal_transfer_operator/data_providers/database/google/bigquery.py#L82-L87

Reply:
This docstring is confusing. We already have a method `is_native_path_available` which is doing the check I believe. Do we need to check here again if optimised path for transfer is available?

Reply:
@pankajkoti 

Following is the complete doc-string which mentions the transfers.
```
Checks if optimised path for transfer between File location to database exists and if it does, it transfers it and returns true else false.
```